lib/compat/wordpress-5.9/edit-site-page.php

```diff
@@ -120,6 +120,8 @@ static function( $classes ) {
 		'defaultTemplateTypes'     => $indexed_template_types,
 		'defaultTemplatePartAreas' => get_allowed_block_template_part_areas(),
 		'__unstableHomeTemplate'   => gutenberg_resolve_home_template(),
+		'__experimentalBlockPatterns'          => WP_Block_Patterns_Registry::get_instance()->get_all_registered(),
+		'__experimentalBlockPatternCategories' => WP_Block_Pattern_Categories_Registry::get_instance()->get_all_registered(),
```

Comment:
Can't you get data via the REST API? Why load this in via settings?

Reply:
The REST API doesn't know all registered patterns. If they are registered in the `current_screen` or the `admin_init` hook, these hooks are not run while serving a REST endpoint. See the PR description and the #40736 that this PR attempts to fix.

Reply:
If you register your pattern in correctly or weirdly, that is on you as a developer. I would strongly recommend asyncly loading pattern via the REST API. This could be a lot of data on the page load, even if it is not used.

Reply:
@spacedmonkey See related conversation about this here: https://github.com/WordPress/gutenberg/issues/40736